lib/chef/provider/apt_repository.rb

```diff
@@ -51,6 +51,19 @@ def load_current_resource
           end
         end
 
+        # make sure we have apt-transport-https installed before we try to
+        # apt-get update with a https repo defined
+        if new_resource.uri =~ /^https/
+          declare_resource(:apt_update, new_resource.name) do
+            ignore_failure true
+            action :periodic
+          end
```

Comment:
feel like this apt_update should probably not go here.  if we want to be defensive around call apt_update for the user before installing packages, then we should do that directly in apt_package.   kind of feels like its trying to backdoor that feature, and either we should really do that in apt_package or never do that for consistency and require the user to always apt_update first.

the `ignore_failure true` is also making me a bit twitchy.  every time i've seen `apt-get update` fail it has been for decent reasons and ignoring it would not have been good.